pages/linux/rename.ul.md

```diff
@@ -0,0 +1,29 @@
+# rename.ul
+
+> Renames multiple files.
+> This is the util-linux version of `rename`. See `rename` for the perl version which is default on Debian.
+> Warning: This command has no safeguards and will overwrite files without prompting.
+
+- Rename files using simple substitutions (substitute 'foo' with 'bar' wherever found):
+
+`rename {{foo}} {{bar}} {{*}}`
+
+- Dry-run - display which renames would occur without performing them:
+
+`rename -vn {{foo}} {{bar}} {{*}}`
+
+- Do not overwrite existing files:
+
+`rename -o {{foo}} {{bar}} {{*}}`
+
+- Change file extensions:
+
+`rename {{.ext}} {{.bak}} *.foo`
+
+- Prepend all files with 'foo':
+
+`rename '' 'foo' {{*}}`
+
+- Rename 'foo1', ..., 'foo10', ..., 'foo100' to 'foo001', ..., 'foo010', ..., 'foo100':
```

Comment:
This command and description is somewhat confusing, and the command is also wrong (the first one should be `rename foo foo0 foo?`, notice only one zero).

I would suggest the following:

    > Rename a group of increasingly numbered files zero-padding the numbers up to 3 digits:

    `rename {{foo}} {{foo0}} {{foo?}} && rename {{foo}} {{foo0}} {{foo??}}`

Reply:
By the way, this becomes even cooler with bash tokens:

    rename bar{,0,?} && rename bar{,0,??}

Reply:
The command is correct. ;)
```
>ls
foo  foo1  foo10  foo100  foo11  foo12  foo2  foo3  foo4  foo5  foo99
>rename foo foo00 foo?
>ls
foo  foo001  foo002  foo003  foo004  foo005  foo10  foo100  foo11  foo12  foo99
>rename foo foo0 foo??
>ls
foo  foo001  foo002  foo003  foo004  foo005  foo010  foo011  foo012  foo099  foo100
```
I'll change it to your suggestion, it's easier to understand. I don't know if I should add the bash tokens? I have never heard of it :o

Reply:
Hmm yeah that shorter example is less clear, let's just go with the long one.